Sorting array via C#

nmamro opened the pull request.

nmamro committed:
Sorting array but now in correct branch

nmamro committed:
Didn't use this

nmamro committed:
Trivial tests

yurii-litvinov commented on Lab1/Task3/Homework/Program.cs:
```diff
@@ -0,0 +1,166 @@
+        {
+            int length = GetLength();
+
+            int[] res = new int[length];
```
Лучше var, тут тип очевиден из правой части присваивания

yurii-litvinov commented on Lab1/Task3/Homework/Program.cs:
```diff
@@ -0,0 +1,166 @@
+            for (int i = 0; i < theArray.Length; ++i)
+            {
+                Console.Write($"{theArray[i]} ");
+            }
```
Это можно было более красиво записать через foreach:
```csharp
        public static void PrintArray(int[] theArray)
        {
            foreach (var element in theArray)
            {
                Console.Write($"{element} ");
            }
        }
```

yurii-litvinov commented on Lab1/Task3/Homework/Program.cs:
```diff
@@ -0,0 +1,166 @@
+
+            int[] testArrayRandom = new int[100];
+
+            Random rand = new Random();
```
Тут тоже правильнее var

yurii-litvinov commented on Lab1/Task3/Homework/Program.cs:
```diff
@@ -0,0 +1,166 @@
+
+            Random rand = new Random();
+
+            for (int i = 0; i < 100; ++i)
```
Магическая константа, к тому же она равна testArrayRandom.Length

yurii-litvinov commented on Lab1/Task3/Homework/Program.cs:
```diff
@@ -0,0 +1,166 @@
+                return false;
+            }
+
+            return true;
```
Тут можно было просто `return TestAscOrder(testArrayRandom);`

yurii-litvinov reviewed (changes requested):
А эту предлагается всё-таки поправить с учётом замечаний к предыдущим задачам

nmamro committed:
Improvements

- Different classes to different files
- Using "var" keyword
- TryParse "out" parameter changes
- Empty line between methods

yurii-litvinov commented on Lab1/Task3/Homework/ArrayUI.cs:
```diff
@@ -0,0 +1,55 @@
+        {
+            var length = GetLength();
+
+            int[] res = new int[length];
```
var

yurii-litvinov commented on Lab1/Task3/Homework/Test.cs:
```diff
@@ -0,0 +1,46 @@
+                return false;
+            }
+
+            int[] testArrayRandom = new int[100];
```
И тут. Тип очевиден из контекста

yurii-litvinov reviewed (approved):
👍

nmamro committed:
Less wars, more vars
